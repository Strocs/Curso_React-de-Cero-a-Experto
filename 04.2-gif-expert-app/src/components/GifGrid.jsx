import { useCamelCase, useFetchGifs } from '../hooks'
import { GifItem } from './'
import PropTypes from 'prop-types'
export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category)
  const { camelCasedText } = useCamelCase(category)
  return (
    <>
      <h3>{camelCasedText}</h3>
      {isLoading && <h2>Cargando...</h2>}
      <div className='card-grid'>
        {images.map(image => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
