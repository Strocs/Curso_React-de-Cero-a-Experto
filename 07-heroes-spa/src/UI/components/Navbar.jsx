import { Link, NavLink, useNavigate } from 'react-router-dom'

export const Navbar = () => {
  const isActive = ({ isActive }) =>
    `nav-item nav-link ${isActive ? 'active' : ''}`

  const navigate = useNavigate()

  const onLogout = () => {
    navigate('/login', {
      replace: true
    })
  }

  return (
    <nav className='navbar navbar-expand-sm navbar-dark bg-dark p-2'>
      <Link className='navbar-brand' to='/'>
        Asociaciones
      </Link>

      <div className='navbar-collapse'>
        <div className='navbar-nav'>
          <NavLink className={isActive} to='/marvel'>
            Marvel
          </NavLink>

          <NavLink className={isActive} to='/dc'>
            DC
          </NavLink>

          <NavLink className={isActive} to='/search'>
            Search
          </NavLink>

          {/* <NavLink className={isActive} to='/hero'>
            Hero
          </NavLink> */}
        </div>
      </div>

      <div className='navbar-collapse collapse w-100 order-3 dual-collapse2 d-flex justify-content-end'>
        <ul className='navbar-nav ml-auto'>
          <span className='nav-item nav-link text-primary m-1'>Ignacio</span>

          <button onClick={onLogout} className='nav-item nav-link btn'>
            Logout
          </button>
        </ul>
      </div>
    </nav>
  )
}
