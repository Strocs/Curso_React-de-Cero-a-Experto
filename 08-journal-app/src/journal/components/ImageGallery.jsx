import { ImageList, ImageListItem } from '@mui/material'

export const ImageGallery = ({ images }) => {
  return (
    <ImageList sx={{ width: '100%', height: 500 }} cols={4} rowHeight={164}>
      {images.map((image, index) => (
        <ImageListItem key={image}>
          <img
            src={`${image}?w=164&h=164&fit=crop&auto=format`}
            srcSet={`${image}?w=164&h=164&fit=crop&auto=format&dpr=2 2x`}
            alt={`Image number ${index} of this note`}
            loading='lazy'
          />
        </ImageListItem>
      ))}
    </ImageList>
  )
}
