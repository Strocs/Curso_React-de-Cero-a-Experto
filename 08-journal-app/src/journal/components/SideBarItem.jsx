import { TurnedInNot } from '@mui/icons-material'
import {
  Grid,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText
} from '@mui/material'
import { useMemo } from 'react'
import { useJournalStore } from '../../stores'

export const SideBarItem = ({ note }) => {
  const setActiveNote = useJournalStore((state) => state.setActiveNote)

  const newTitle = useMemo(() => {
    return note.title.length > 17
      ? note.title.substring(0, 17) + '...'
      : note.title
  }, [note.title])
  return (
    <ListItem key={note.id}>
      <ListItemButton onClick={() => setActiveNote(note)}>
        <ListItemIcon>
          <TurnedInNot />
        </ListItemIcon>
        <Grid container>
          <ListItemText primary={newTitle} />
          <ListItemText secondary={note.body} />
        </Grid>
      </ListItemButton>
    </ListItem>
  )
}
