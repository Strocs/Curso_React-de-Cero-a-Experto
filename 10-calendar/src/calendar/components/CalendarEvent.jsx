export const CalendarEvent = ({ event }) => {
  const { title, user } = event || null

  return (
    <>
      <strong>{title}</strong>
      <span> - {user.name}</span>
    </>
  )
}
