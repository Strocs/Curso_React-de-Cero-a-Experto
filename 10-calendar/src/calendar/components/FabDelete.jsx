import { useCalendarStore, useUiStore } from '../../hooks'

export const FabDelete = () => {
  const { startDeletingEvent, hasEventSelected } = useCalendarStore()
  const { isModalClosed } = useUiStore()
  const handleDelete = async () => {
    await startDeletingEvent()
  }

  return (
    <button
      onClick={handleDelete}
      className='btn btn-danger fab-danger'
      style={{
        display: hasEventSelected && isModalClosed ? '' : 'none'
      }}>
      <i className='fas fa-trash-alt fa-xs' />
    </button>
  )
}
