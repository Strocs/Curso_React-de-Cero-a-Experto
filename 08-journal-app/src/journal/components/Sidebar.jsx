import { Box, Divider, Drawer, List, Toolbar, Typography } from '@mui/material'
import { useJournalStore } from '../../stores'
import { useAuthStore } from '../../stores/useAuthStore'
import { SideBarItem } from './SideBarItem'

export const Sidebar = ({ drawerWidth = 240 }) => {
  const { displayName } = useAuthStore((state) => state.userAuth)
  const notes = useJournalStore((state) => state.notes)

  return (
    <Box
      component='nav'
      sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}
    >
      <Drawer
        variant='permanent'
        open
        sx={{
          display: { xs: 'block' },
          '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth }
        }}
      >
        <Toolbar>
          <Typography variant='h6' noWrap component={'div'}>
            {displayName}
          </Typography>
        </Toolbar>
        <Divider />

        <List>
          {notes.map((note) => (
            <SideBarItem note={note} key={note.id} />
          ))}
        </List>
      </Drawer>
    </Box>
  )
}
