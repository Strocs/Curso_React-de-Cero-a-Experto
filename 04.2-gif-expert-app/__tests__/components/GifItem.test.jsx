import { render, screen } from '@testing-library/react'
import { describe, test, expect } from 'vitest'
import { GifItem } from '../../src/components/GifItem'

describe('Pruebas en <GifItem />', () => {
  const title = 'Imagen de tio1.jpg'
  const url = 'https://tio1.jpg/'

  test('should match with the snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('should show the image with given url and alt', () => {
    render(<GifItem title={title} url={url} />)
    // expect(screen.getByRole('img').src).toBe(url)
    // expect(screen.getByRole('img').alt).toBe(title)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('should show title in component', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  })
})
