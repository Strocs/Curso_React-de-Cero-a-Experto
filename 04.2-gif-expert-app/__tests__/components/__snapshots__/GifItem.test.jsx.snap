// Vitest Snapshot v1

exports[`Pruebas en <GifItem /> > should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Imagen de tio1.jpg"
      src="https://tio1.jpg/"
    />
    <p>
      Imagen de tio1.jpg
    </p>
  </div>
</div>
`;
